icu4j/main/classes/core/src/com/ibm/icu/util/MeasureUnit.java

```diff
@@ -1471,6 +1907,29 @@ private Object writeReplace() throws ObjectStreamException {
         return new MeasureUnitProxy(type, subType);
     }
 
+    /**
+     *
+     * @return this object as a SingleUnitImpl.
+     * @throws UnsupportedOperationException if this object could not be converted to a single unit.
+     */
+    private SingleUnitImpl getSingleUnitImpl() {
+        if (measureUnitImpl == null) {
+            return MeasureUnitImpl.forIdentifier(getIdentifier()).getSingleUnitImpl();
+        }
+
+        return measureUnitImpl.getSingleUnitImpl();
+    }
+
+    /**
+     *
+     * @return this object in a MeasureUnitImpl form.
+     */
+    private MeasureUnitImpl getCopyOfMeasureUnitImpl() {
```

Comment:
You should have to versions of this function: one that returns a copy (like this), and one that returns a read-only reference (without the `.clone()`) if the impl is already present in the MeasureUnit.

Reply:
yes, I have added `getMayBeReferenceOfMeasureUnitImpl`